magpie/api/management/user/user_views.py

```diff
@@ -124,35 +124,26 @@ def delete_user_view(request):
 @view_config(route_name=s.UserGroupsAPI.name, request_method="GET", permission=MAGPIE_CONTEXT_PERMISSION)
 def get_user_groups_view(request):
     """
-    List all groups a user belongs to. Any pending group is not included in this list.
+    List all groups a user belongs to.
+    Groups can be of active or pending type, or include both types depending of input arguments.
     """
     user = ar.get_user_matchdict_checked_or_logged(request)
-    group_names = uu.get_user_groups_checked(user, request.db)
-    return ax.valid_http(http_success=HTTPOk, content={"group_names": group_names},
-                         detail=s.UserGroups_GET_OkResponseSchema.description)
-
-
-@s.UserPendingGroupsAPI.get(schema=s.UserPendingGroups_GET_RequestSchema, tags=[s.UsersTag],
-                            response_schemas=s.UserPendingGroups_GET_responses, api_security=s.SecurityAuthenticatedAPI)
-@s.LoggedUserPendingGroupsAPI.get(schema=s.UserPendingGroups_GET_RequestSchema, tags=[s.LoggedUserTag],
-                                  response_schemas=s.LoggedUserPendingGroups_GET_responses,
-                                  api_security=s.SecurityAuthenticatedAPI)
-@view_config(route_name=s.UserPendingGroupsAPI.name, request_method="GET", permission=MAGPIE_CONTEXT_PERMISSION)
-def get_user_pending_groups_view(request):
-    """
-    List all groups where the user's membership is pending, because of required terms and conditions acceptation.
-    """
-    user = ar.get_user_matchdict_checked_or_logged(request)
-    tmp_tokens = TemporaryToken.by_user(user).filter(TemporaryToken.operation == TokenOperation.GROUP_ACCEPT_TERMS)
-    pending_group_names = [tmp_token.group.group_name for tmp_token in tmp_tokens]
+    group_type = ar.get_multiformat_body(request, "group_type", default=UserGroupType.ACTIVE_USERGROUPS.value)
```

Comment:
Requires validation of the value. 
This also doesn't fetch from the query string, but rather the request body (not available for GET).

I think tests are in order also to ensure these cases: 
- old/default listing is returned if nothing/"active" is provided
- only pending groups are returned when asking for it
- both are returned when "all"

Reply:
If that works for you, I will complete those 2 tests ([test_GetUserGroups](https://github.com/Ouranosinc/Magpie/blob/9deeda69e50bbc4f06ce1f3670f09a5bfc6cafae/tests/interfaces.py#L4092), [test_GetGroupUsers](https://github.com/Ouranosinc/Magpie/blob/9deeda69e50bbc4f06ce1f3670f09a5bfc6cafae/tests/interfaces.py#L4475)) by adding all the possible cases (all, active, pending) of the GET queries.

Reply:
I would rather have separate tests that have explicit queries so the current ones remain simple.

Reply:
I am not sure I understand. By separate tests, you mean one test for each case? 
So basically :
4 different test cases (all, pending, active, and no specified status)
x 2 query types (for GetUserGroups and GetGroupUsers)
So 8 tests in total.
The 2 existing ones I linked in my last comment would correspond to 2 of those 8 cases.

I think it would make more sense to keep all 4 different test cases in one single test for each query type. So, we keep just the 2 original tests.
This way, I can setup the user/groups just one time. For example, for the GetUserGroups query, I just have to add the user to a group with terms, and to a group without terms, and then check with the 4 GET query status argument cases, if the results are as expected.

Reply:
I pushed the 2 completed tests, which each include the 4 status test cases, let me know what you think.

Reply:
With separate tests it is easier to identify if only one case fails for some underlying reason or all of them are invalid. 
When combined in one test, the first error fails the test, but we don't get the details of (possible) other failing cases.

You can make an "util" method that does the same procedure with an input query and expected members to reduce duplication of code for all combinations.